core/object/worker_thread_pool.cpp

```diff
@@ -361,6 +361,26 @@ bool WorkerThreadPool::is_task_completed(TaskID p_task_id) const {
 	return (*taskp)->completed;
 }
 
+bool WorkerThreadPool::clear_task_if_completed(TaskID p_task_id) {
+	task_mutex.lock();
+	Task **taskp = tasks.getptr(p_task_id);
+	if (!taskp) {
+		task_mutex.unlock();
```

Comment:
If the task has been already waited on, it will raise an error.

Reply:
Oh yeah I guess that lock isn't needed huh

Reply:
Just removed the locks!